crates/cairo-profiler/src/trace_reader/function_trace_builder/inlining.rs

```diff
@@ -0,0 +1,74 @@
+use std::cmp::max;
+
+use cairo_lang_sierra::program::StatementIdx;
+use itertools::Itertools;
+
+use crate::sierra_loader::StatementsFunctionsMap;
+use crate::trace_reader::function_name::FunctionName;
+use crate::trace_reader::sample::{FunctionCall, InternalFunctionCall};
+
+pub(super) fn build_original_call_stack_with_inlined_calls(
+    sierra_statement_idx: StatementIdx,
+    maybe_statements_functions_map: Option<&StatementsFunctionsMap>,
+    current_function_call_stack: Vec<FunctionCall>,
+) -> Vec<FunctionCall> {
+    let maybe_original_call_stack_postfix = maybe_statements_functions_map
```

Comment:
It indeed should have been a suffix. It is there because the call stack from mappings is always a suffix of the current call stack (if the mappings and our logic are correct)